lib/akami/wsse/signature.rb

```diff
@@ -151,6 +193,12 @@ def body_digest
         Base64.encode64(OpenSSL::Digest::SHA1.digest(body)).strip
       end
 
+      def timestamp_digest
+        return nil unless timestamp
```

Comment:
Should tis be `return nil unless self.timestamp`.  It seems that `timestamp` refers to the variable in the scope of the method due tot he assignment on the next line.